pkg/pgmodel/querier/querier.go

```diff
@@ -237,15 +243,19 @@ func (q *pgxQuerier) queryMultipleMetrics(filter metricTimeRangeFilter, cases []
 	// Generate queries for each metric and send them in a single batch.
 	for i, metric := range metrics {
 		//TODO batch getMetricTableName
-		tableName, err := q.getMetricTableName(metric)
+		tableSchema, tableName, seriesTable, err := q.getMetricTableName(schemas[i], metric)
 		if err != nil {
 			// If the metric table is missing, there are no results for this query.
 			if err == errors.ErrMissingTableName {
 				continue
 			}
 			return nil, nil, err
 		}
+
 		filter.metric = tableName
+		filter.schema = tableSchema
```

Comment:
should we verify this is the data schema and error if not?

Reply:
The SQL query uses `series` table to fetch the metric IDs which means it will only fetch non-view metric IDs.

Reply:
yes but then wouldn't that be a good safety check here?